include/LeMonADE/analyzer/AnalyzerRadiusOfGyration.h

```diff
@@ -59,227 +44,229 @@ along with LeMonADE.  If not, see <http://www.gnu.org/licenses/>.
+ * of this data to disk in the format 
+ * mcs <Rg2_x> <Rg2_y> <Rg2_z>  <Rg2_tot>
+ * The default output filename is Rg2TimeSeries.dat and it can be changed by argument
+ * to the constructor or by using the setter function provided.
+ * If a more sophisticated grouping of monomers into groups is required, one can 
+ * also write a new analyzer, inheriting from AnalyzerRadiusOfGyration, and 
+ * overwriting the initialize function. 
  */
 template < class IngredientsType > class AnalyzerRadiusOfGyration : public AbstractAnalyzer
 {
-	typedef typename IngredientsType::molecules_type molecules_type;
 	
+private:
+	//! typedef for the underlying container holding the monomers
+	typedef typename IngredientsType::molecules_type molecules_type;
+	//! reference to the complete system
 	const IngredientsType& ingredients;
-	
+	//! Rg2 is calculated for the groups in this vector
 	std::vector<MonomerGroup<molecules_type> > groups;
+	//! timeseries of the Rg^2. Components: [0]-> Rg^2_x, [1]->Rg^2_y, [2]->Rg^2_z, [3]:Rg^2_tot
+	std::vector< std::vector<double> > Rg2TimeSeries;
+	//! vector of mcs times for writing the time series
+	std::vector<double> MCSTimes;
+	//! max length of internal buffer for each coordinate, before saving to disk
+	uint32_t bufferSize;
+	//! name of output files are outputFilePrefix_averages.dat and outputFilePrefix_timeseries.dat
+	std::string outputFile;
+	//! flag used in dumping time series output
+	bool isFirstFileDump;
+	//! save the current values in Rg2TimeSeriesX, etc., to disk
+	void dumpTimeSeries();
+	//! calculate the Rg squared of the monomer group
+	VectorDouble3 calculateRg2Components(const MonomerGroup<molecules_type>& group) const;
 	
-	TimeSeries<VectorDouble3>  Rg2TimeSeries;
-	std::vector<Moments<double> > Rg_squaredX,Rg_squaredY,Rg_squaredZ,Rg_squared;
-	
-	//! mcs below this number are not considered in analysis
-	uint64_t lowerLimit;
-	
-	//! prefix for output filename
-	std::string outputFilePrefix;
 public:
   
-	/*
-	 * there are three different constructors: 
-	 * - the first calculates the Rg^2 of the complete system
-	 * - the second takes a monomer group as additional argument and 
-	 *   calculates the average Rg^2 of the monomers in this group
-	 * - the third takes a std::vector of monomer groups and calculates
-	 *   the Rg^2 of all molecules in this group. The time series output is
-	 *   then the average Rg^2, averaged over both time and groups.
-	 * */
-	AnalyzerRadiusOfGyration(const IngredientsType& ing,std::string filenamePrefix="",uint64_t lowerMcsLimit=0);
+	//! constructor
 	AnalyzerRadiusOfGyration(const IngredientsType& ing,
-				 const MonomerGroup<molecules_type>& g,
-			  std::string filenamePrefix="",
-			  uint64_t lowerMcsLimit=0);
-	AnalyzerRadiusOfGyration(const IngredientsType& ing,
-				 const std::vector<MonomerGroup<molecules_type> >& g,
-			  std::string filenamePrefix="",
-			  uint64_t lowerMcsLimit=0);
+			  std::string filename=std::string("Rg2TimeSeries.dat"));
```

Comment:
I would prefere to change 
std::string filename=std::string("Rg2TimeSeries.dat")
to 
std::string filename("Rg2TimeSeries.dat")
which calls a constructor only, and does not need a = operator call. Of cause this might be optimized by compiler, but it also looks better ;o)